### PR TITLE
Use lib instead of ar on Windows

### LINKED ISSUE #384: Windows Command Prompt Can't Create Archives
Running in a Windows Command Prompt, I get the error:

```
'ar' is not recognized as an internal or external command,
operable program or batch file.
```

which is not surprising, but I'm not sure what the correct command would be on Windows for creating archives for libraries.

### PR DESCRIPTION
**Pull request by everythingfunctional**
If `ar` isn't available, use `lib` to create static library files. Only do this check on Windows (or Unknown) systems. It's almost certainly safe to assume `ar` exists on all other systems.

Fix #384

### CODE REVIEW

File: src/fpm.f90 [hunk 1]
```diff
@@ -62,6 +62,23 @@ subroutine build_model(model, settings, package, error)
...
+
+                call execute_command_line("ar --version", exitstat=estat)
+                if (estat /= 0) then
+                    model%archiver = "lib /OUT:"
+                else
+                    model%archiver = "ar -rs "
+                end if
+            end block
+        end if
+    end associate
```
Comment (awvwgk) [inline src/fpm.f90:80]: Why not make it a function?

**Review by LKedward — APPROVED** [review]
Thanks @everythingfunctional. I think this looks good for now to get improved support on Windows.

---
In the future, this functionality may be better placed in [`fpm_compiler.f90`](https://github.com/fortran-lang/fpm/blob/a540c83d9e303acd6fece95927c49fa8d2565366/src/fpm_compiler.f90) in order to invoke compiler-specific library tools (_e.g._ `gcc-ar`, `xiar`, `xilib`) needed for LTO.

File: src/fpm.f90 (hunk 1 above)
Comment (everythingfunctional) [reply src/fpm.f90:80]: Good point. I'll extract it into a function, `get_archiver`.

**Comment by everythingfunctional** [comment]
Having addressed the one comment, and with another approval, I'm going to go ahead and merge this. Hopefully it's not too difficult for #449 and #451 to incorporate this.

**Pull request merged**